rcl_action/src/rcl_action/graph.c

```diff
@@ -154,6 +154,10 @@ rcl_action_get_client_names_and_types_by_node(
   rcl_ret_t nat_fini_ret = rcl_names_and_types_fini(&topic_names_and_types);
   if (RCL_RET_OK != nat_fini_ret) {
     ret = rcl_names_and_types_fini(action_names_and_types);
+    if (RCL_RET_OK != ret) {
+      RCL_SET_ERROR_MSG(
```

Comment:
Won't this "overwrite" the existing error and cause extra error messages from the error handling logic in rcutils? E.g. `"This error state is being overwritten"`.

I think it would be better to instead call `RCUTILS_SAFE_FWRITE_TO_STDERR`, like we do in similar error-during-error cases (e.g. https://github.com/ros2/rcl/pull/834).

Reply:
I wasn't sure how to handle this case; thanks for the info.  I'll switch to that.